pg_search/src/index/search.rs

```diff
@@ -461,6 +540,102 @@ impl<'de> Deserialize<'de> for SearchIndex {
         let mut underlying_index =
             Index::open_in_dir(tantivy_dir_path).expect("failed to open index");
 
+        // #[derive(Clone)]
+        // struct AdvisoryLockDirectory {
+        //     index_oid: u32,
+        //     mmap: tantivy::directory::MmapDirectory,
+        // }
```

Comment:
here's the custom Directory impl that also takes an advisory lock.

Don't ask me why it's all commented out -- I must have been in the middle of something when I had to stash this and now I don't remember.